services/dkg/pedersen/controller/mod.go

```diff
@@ -100,22 +110,47 @@ func (m controller) OnStart(ctx cli.Flags, inj node.Injector) error {
 		return xerrors.Errorf("failed to resolve *cosipbft.Service")
 	}
 
-	dkg, pubkey := pedersen.NewPedersen(no, true, srvc, rosterFac)
+	var dkgMap kv.DB
+	err = inj.Resolve(&dkgMap)
+	if err != nil {
+		return xerrors.Errorf("failed to resolve kv.DB")
+	}
+
+	dkg := pedersen.NewPedersen(no, srvc, rosterFac)
+
+	// Use dkgMap to fill the actors map
+	err = dkgMap.View(func(tx kv.ReadableTx) error {
+		bucket := tx.GetBucket([]byte(DKGMAP))
+		if bucket == nil {
+			return nil
+		}
 
-	pubkeyBuf, err := pubkey.MarshalBinary()
+		return bucket.ForEach(func(electionIDBuf, handlerDataBuf []byte) error {
+
+			handlerData := pedersen.HandlerData{}
+			err = json.Unmarshal(handlerDataBuf, &handlerData)
+			if err != nil {
+				return err
+			}
+
+			_, err = dkg.NewActor(electionIDBuf, handlerData)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+	})
 	if err != nil {
-		return xerrors.Errorf("failed to encode pubkey: %v", err)
+		return xerrors.Errorf("database read failed: %v", err)
 	}
 
 	inj.Inject(dkg)
-	inj.Inject(pubkey)
-
-	dela.Logger.Info().
-		Hex("public key", pubkeyBuf).
-		Msg("perdersen public key")
 
 	rosterKey := [32]byte{}
-	evoting.RegisterContract(exec, evoting.NewContract(evotingAccessKey[:], rosterKey[:], access, dkg, rosterFac))
+	// TODO This shouldn't take dkg as input, but rather the map?
```

Comment:
true, the smart contract should only receive an interface that defines `GetActor(electionID []byte) (Actor, boolean)`.